gateway/src/apicast/proxy.lua

```diff
@@ -225,7 +225,13 @@ function _M:rewrite(service, context)
 
   ngx.var.secret_token = service.secret_token
 
-  local credentials, err = service:extract_credentials()
+  -- Another policy might have already extracted the creds.
+  local credentials = service.extracted_credentials
```

Comment:
Because `.credentials` already exists in service and it has a different kind of information. It contains the location for credentials (query, headers, etc.) and also the name used to look for the different types of keys:

https://github.com/3scale/apicast/blob/master/gateway/src/apicast/configuration.lua#L106

Reply:
Oh boy. This would definitely would benefit from some cleanup.

Reply:
What about storing this in`context.credentials` ? Or better we do that when we have proper `Credentials` class/object ?

Reply:
I would wait until we extract the `Credentials` class.